### PR TITLE
Update TextModels to work with Flux @0.12 and Zygote

### LINKED ISSUE #1: Update TextModels for supporting flux-zygote
Update TextAnalysis to Flux@0.10.0 with Zygote.

- Without compromising on performance, Zygote supports the full flexibility and dynamism of the Julia language, including control flow, recursion, closures, structs, dictionaries, and more.

- DiffEqFlux and NeuralNetDiffEq already moved to zygote.

- On the other hand Tracker is alive but is no longer heavily maintained so it is better to use Zygote

### PR DESCRIPTION
**Pull request by AdarshKumar712**
This PR updates the current version of the models available with TextModels.jl to work with Flux@0.12 and Zygote. Will fix #1 and #19  

Models updated so far:
- [x] CRF
- [x] NER
- [x] POS
- [x] Sentiment.jl (no changes)
- [x] averagePerceptronTagger.jl (no changes)

Also updated Project.toml

Work to be done:
- [x] Update ULMFit model
- [x] Update Tests
- [x] Update Docs (Need Suggestions on this!)

PS. Currently I have commented out the section of code belonging to ULMFit Model. Will uncomment it once its updated.

Please let me know if there are some mistakes. 

cc @aviks 

<b>Update</b>: Updated the tests and checked as well. For ULMFiT model, I am unable to verify the trainable part as the model is running very slow on my system. Rest of the models are working fine.
<b>Update -2 </b>: Documentation for some models, not available earlier is also moved from TextAnalysis.jl

### CODE REVIEW

**Comment by sambitdash** [comment]
I was working on fixing this. But didn't realise @AdarshKumar712 was working on it. So I have submitted all that I have done as part of PR #21. 

In #21, I have made some changes to ULMFit, CRF, NER and POS. Some test cases in NER, POS are broken and have been marked as well. I feel there may be a retraining needed for NER and POS as the model is changed in Flux for the LSTM. The precomputed weights are not good enough. The retrainable parameter logic in ULMFit is probably broken. That may not be needed as well. One can provide the `@functor` macro the relevant params to set the right retrainable parameters. 

@AdarshKumar712 feel free to use any code from #21 if you find it of use.

**Comment by AdarshKumar712** [comment]
I will surely look into the changes you made in your PR. Thanks a lot @sambitdash!

**Comment by AdarshKumar712** [comment]
Currently, CI build is giving a dependency error while precompiling because of `CorpusLoaders.jl` (used for ULMFit module). I have created an issue https://github.com/JuliaText/CorpusLoaders.jl/issues/43 regarding the same. Once the issue resolves, hopefully CI build will work properly. And accordingly, I will also update this PR.

**Comment by AdarshKumar712** [comment]
I am thinking to do the documentation update separately. Other than that, the rest is updated. @aviks can you once have a look?

**Comment by aviks** [comment]
So @AdarshKumar712 is this ready to merge? There is a test failure in nightly, but I think that is an upstream issue.

**Comment by AdarshKumar712** [comment]
Yes, this is ready for now. If there will be any issues, that can be accommodated in future. On a side note, the ULMFiT model is a bit way too slow. So wasn't able to completely check for the Pretraining and fine-tuning. But rest works fine.

File: src/ULMFiT/custom_layers.jl [hunk 1]
```diff
@@ -60,17 +60,17 @@ function WeightDroppedLSTMCell(in::Integer, out::Integer, p::Float64=0.0;
...
     cell = WeightDroppedLSTMCell(
-        param(init(out*4, in)),
-        param(init(out*4, out)),
-        param(init(out*4)),
-        param(zeros(Float32, out)),
-        param(zeros(Float32, out)),
+        init(out*4, in),
+        init(out*4, out),
+        init(out*4),
+        reshape(zeros(Float32, out),out, 1),
```
Comment (tejasvaidhyadev) [inline src/ULMFiT/custom_layers.jl]: reshape(zeros(Float32, out), out, 1),

File: src/ULMFiT/custom_layers.jl [hunk 2]
```diff
@@ -106,8 +109,18 @@ julia> wd = WeightDroppedLSTM(4, 5, 0.3);
 function WeightDroppedLSTM(a...; kw...)
     cell = WeightDroppedLSTMCell(a...;kw...)
     hidden = (cell.h, cell.c)
-    return Flux.Recur(cell, hidden, hidden)
+    return Flux.Recur(cell, hidden)
+end
+
+# over definition for reset! to work with pretrained model
```
Comment (tejasvaidhyadev) [inline src/ULMFiT/custom_layers.jl]: try using DocString

File: src/ULMFiT/fine_tune_lm.jl [hunk 3]
```diff
@@ -50,32 +50,28 @@ This function contains main training loops for fine-tuning the language model.
 To use this funciton, an instance of LanguageModel and a data loader is needed.
```
Comment (tejasvaidhyadev) [inline src/ULMFiT/fine_tune_lm.jl]: function* typo

**Review by tejasvaidhyadev — CHANGES_REQUESTED** [review]

**Comment by tejasvaidhyadev** [comment]
looks good to me

**Comment by AdarshKumar712** [comment]
Updated as per the suggestions. Hopefully, now it's ready to merge.
cc @aviks @tejasvaidhyadev

**Review by tejasvaidhyadev — APPROVED** [review]
Looks good to me

**Comment by tejasvaidhyadev** [comment]
@aviks 
should we merge it?

**Comment by aviks** [comment]
Ok, here goes...

**Pull request merged**